discovery/src/main/java/com/ibm/watson/developer_cloud/discovery/v1/Discovery.java

```diff
@@ -76,379 +45,803 @@
+  }
+
+  /**
+   * Add a document.
+   *
+   * Add a document to a collection with optional metadata and optional configuration.   Set the `Content-Type`
+   * header on the request to indicate the media type of the document. If the `Content-Type` header is missing or is
+   * one of the generic media types (for example,  `application/octet-stream`), then the service attempts to
+   * automatically detect the document's media type.       * The configuration to use to process the document can be
+   * provided by using the `configuration_id` query parameter.       * The `version` query parameter is still
+   * required.    * Returns immediately after the system has accepted the document for processing.    * The user must
+   * provide document content, metadata, or both. If the request is missing both document content and metadata, it is
+   * rejected.       * The user can set the `Content-Type` parameter on the `file` part to indicate the media type of
+   * the document. If the `Content-Type` parameter is missing or is one of the generic media types (for example,
+   * `application/octet-stream`), then the service attempts to automatically detect the document's media type.
+   *
+   * @param environmentId   the ID of your environment
+   * @param collectionId    the ID of your collection
+   * @param configurationId The ID of the configuration to use to process the document. If the `configuration` form
+   *                        part is also provided (both are present at the same time), then request will be rejected.
+   * @param file            The content of the document to ingest.The maximum supported file size is 50 megabytes.
+   *                        Files larger than 50 megabytes is rejected.
+   * @param fileMediaType   the media type of file.
+   * @param metadata        If you're using the Data Crawler to upload your documents, you can test a document
+   *                        against the type of metadata that the Data Crawler might send. The maximum supported
+   *                        metadata file size is 1 MB. Metadata parts larger than 1 MB are rejected. Example:  ``` {
+   *                        "Creator": "Johnny Appleseed",   "Subject": "Apples" } ```
+   * @param configuration   The configuration to use to process the document. If this part is provided, then the
+   *                        provided configuration is used to process the document. If the `configuration_id` is also
+   *                        provided (both are present at the same time), then request is rejected. The maximum
+   *                        supported configuration size is 1 MB. Configuration parts larger than 1 MB are rejected.
+   *                        See the `GET /configurations/{configuration_id}` operation for an example configuration.
+   * @return the {@link DocumentAccepted} with the response
+   */
+  public ServiceCall<DocumentAccepted> addDocument(String environmentId, String collectionId, String configurationId,
```

Comment:
to me this seems to be super cumbersome in java world + against the rule of 3 parameters that you stated would be collapsed into an options object